### DISCUSSION TITLE
About the twit of UTM

### DISCUSSION THREAD

**Discussion by Suchan0127** [utmapp/UTM#3511] [UTM Help (iOS)]
https://mobile.twitter.com/UTMapp/status/1440055183296262147?cxt=HHwWhsCrleejjfwnAAAA
on this thread, UTM said that he was able to emulate Windows 7 with 4GB RAM with iPad M1.
As a same M1 user AND a UTM user, I tried to allocate 4GB RAM but failed (Worked with 4GB RAM and 512MB of JIT Memory). I thought It was a problem of new UTM, so I tried UTM 2.2.4 (Version mentioned on the ghread). However It didn’t work as same.
Any solution that I can also allocate 4GB RAM? (I also have Increased-Memory Limit entitlement too, Imma Develper Account Owner)

**Comment by osy** [comment]
I think the RAM limit was changed in iOS 15. iOS 15 lets you use less ram than iOS 14 even with the entitlements.

**Reply by Suchan0127** [reply]
So… Were you on iOS 14 when you uploaded the twitt?
Also I thought Increased RAM limit was only available on M1 devices and iOS/iPadOS 15

**Reply by osy** [reply]
Good point I think I was on iOS 15 beta

**Reply by Suchan0127** [reply]
Mhm. Makes sense.
Sorry that I’m making the problem too personally but well…
1. I attempted instlling MacOS Snow Leopard on UTM (iOS)
Should I emulate it w/ x86_64 or PPC?

2. Purple light flashes and iPad restart when OpenGPU Acceleration true. Any solution?
3. I found it out UTM VM only support Local Network. How can I use WIFI? If not, can you update the app so it support WIFI network?

**Comment by kkebo** [comment]
Did you add not only the `com.apple.developer.kernel.increased-memory-limit` entitlement but also the `com.apple.developer.kernel.extended-virtual-addressing` entitlement?

If not, the app will be killed by the OS when the app uses around 4-5GB RAM.

To build UTM with that entitlement enabled, you need to enable "Extended Virtual Address Space" manually in https://developer.apple.com/account/resources/identifiers/.

![image](https://user-images.githubusercontent.com/601636/150524216-9d928577-b93c-489b-8c33-e9f19a34aeb0.png)

I added that entitlement in https://github.com/utmapp/UTM/pull/3318 and this change has been included in UTM 3.0.1 or later.

Here is the screenshots when I run a VM with 10GB RAM:

https://twitter.com/c10um0/status/1479830519424888835?s=21

**Reply by Suchan0127** [reply]
Thx for advice. I should try it at home!
Also, I believe 10GB RAM was only possible on your device.
I wonder how much RAM does your iPad have.
Could I ask your iPad RAM?

**Reply by kkebo** [reply]
My one has 16 GB RAM, but the actual amount available is 12245 MB.

![IMG_0863](https://user-images.githubusercontent.com/601636/150527489-897b974f-7eac-4474-98ac-eae4daee175b.png)

Even if your M1 iPad Pro is an 8 GB RAM model, adding the `com.apple.developer.kernel.extended-virtual-addressing` entitlement will at least prevent it from crashing at 4-5 GB of usage.

**Reply by Suchan0127** [reply]
So you mean if I add com.apple.developer.kernel.extended-virtual-addressing entitlememt I can allocate 4GB RAM on my iPad right?

**Reply by kkebo** [reply]
Yes.

**Reply by Suchan0127** [reply]
With 1024MB of JIT cache?

**Reply by kkebo** [reply]
Yes, that's right.

**Reply by Suchan0127** [reply]
Hi again. I attemped to install UTM with com.apple.developer.kernel.extended-virtual-addressing, but unfortunately failed. Is it correct that I can use more RAM in UTM than com.apple.developer.kernel.increased-memory-limit entitlement? I didn’t see any difference between turning both on and only com.apple.developer.kernel.increased-memory-limit on. Any solution?

**Reply by Suchan0127** [reply]
Sorry, actually worked. Really thanks!!!

**Reply by Suchan0127** [reply]
Sorry again, I found that app keeps crash after I alloated 4GB of RAM after about a minute or so. Any solution??

**Reply by kkebo** [reply]
> I attemped to install UTM with com.apple.developer.kernel.extended-virtual-addressing, but unfortunately failed.

Did you do the following thing?

> To build UTM with that entitlement enabled, you need to enable "Extended Virtual Address Space" manually in https://developer.apple.com/account/resources/identifiers/.

I would also recommend reading the comment in the following thread.

https://developer.apple.com/forums/thread/685084?answerId=695278022#695278022
